source/frontend/src/features/mapSideBar/acquisition/tabs/expropriation/ExpropriationTabContainer.tsx

```diff
@@ -1,26 +1,55 @@
-import { useContext } from 'react';
+import { useCallback, useContext, useEffect, useState } from 'react';
 
 import { SideBarContext } from '@/features/mapSideBar/context/sidebarContext';
+import { useAcquisitionProvider } from '@/hooks/repositories/useAcquisitionProvider';
 import { Api_AcquisitionFile } from '@/models/api/AcquisitionFile';
+import { Api_ExpropriationPayment } from '@/models/api/ExpropriationPayment';
 
 import { IExpropriationTabContainerViewProps } from './ExpropriationTabContainerView';
 
-export interface IExpropriationTabContainer {
+export interface IExpropriationTabContainerProps {
   acquisitionFile: Api_AcquisitionFile;
   View: React.FunctionComponent<IExpropriationTabContainerViewProps>;
 }
 
-export const ExpropriationTabContainer: React.FunctionComponent<IExpropriationTabContainer> = ({
-  View,
-  acquisitionFile,
-}) => {
+export const ExpropriationTabContainer: React.FunctionComponent<
+  React.PropsWithChildren<IExpropriationTabContainerProps>
+> = ({ View, acquisitionFile }) => {
   const { fileLoading } = useContext(SideBarContext);
+  const [form8s, setForm8s] = useState<Api_ExpropriationPayment[]>([]);
+
+  const {
+    getAcquisitionFileForm8s: { execute: getAcquisitionFileForm8s, loading: loadingForm8s },
+  } = useAcquisitionProvider();
+
+  const fetchForms = useCallback(async () => {
+    if (acquisitionFile.id) {
+      var retrieved = await getAcquisitionFileForm8s(acquisitionFile.id);
+      if (retrieved) {
+        setForm8s(retrieved);
+      }
+    }
+  }, [acquisitionFile.id, getAcquisitionFileForm8s, setForm8s]);
+
+  useEffect(() => {
+    fetchForms();
+  }, [fetchForms]);
+
+  const handleForm8Deleted = (form8Id: number) => {
+    setForm8s(current => current.filter(form => form.id !== form8Id));
```

Comment:
would recommend removing this, and just requerying the form8s after a deletion. this runs the risk of concurrency related errors.

Reply:
updated